plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3RetryingInputStream.java

```diff
@@ -151,24 +180,36 @@ private void reopenStreamOrFail(IOException e) throws IOException {
         if (failures.size() < MAX_SUPPRESSED_EXCEPTIONS) {
             failures.add(e);
         }
-        try {
-            Streams.consumeFully(currentStream);
-        } catch (Exception e2) {
-            logger.trace("Failed to fully consume stream on close", e);
-        }
+        maybeAbort(currentStream);
         IOUtils.closeWhileHandlingException(currentStream);
-        currentStream = openStream();
+        openStream();
     }
 
     @Override
     public void close() throws IOException {
+        maybeAbort(currentStream);
         try {
-            Streams.consumeFully(currentStream);
+            currentStream.close();
+        } finally {
+            closed = true;
+        }
+    }
+
+    /**
+     * Abort the {@link S3ObjectInputStream} if it wasn't read completely at the time this method is called,
+     * suppressing all thrown exceptions.
+     */
+    private void maybeAbort(S3ObjectInputStream stream) {
+        if (eof) {
```

Comment:
`if (eof || start + currentOffset == currentStreamLastOffset) {` and drop the conditional from the `try {` since both cases mean the same to us?